docs/content/preview/develop/quality-of-service/write-heavy-workloads.md

```diff
@@ -23,66 +24,59 @@ The idea is to slow down incoming writes to the speed that the database can hand
+
+If WAL writes are slow, then writes experience higher latency which creates a natural form of admission control. The frequency of data getting synced to disk is controlled by the `durable_wal_write` flag. Note that `fsync` is disabled by default as YugabyteDB is expected to run in a Replication Factor 3 mode where each shard/tablet is replicated onto 3 independent fault domains such as hosts, availability zones, regions, or clouds. If enabled, every single write into DocDB, YugabyteDB's underlying document store, will be synchronized to disk before its execution is deemed successful. So, although there will be an increase in safety, there will be a hit to performance.
 
 ### Limited disk IOPS or bandwidth
-In many cloud environments, the disk IOPS and/or the network bandwidth is rate-throttled. Holistically, if there are such disk constraints (limits on iops / bandwidth), that results in a back-pressure across all disk writes in the system, and would therefore manifest as one of the above conditions.
 
+In many cloud environments, the disk IOPS and/or the network bandwidth is rate-throttled. Holistically, if there are such disk constraints (limits on IOPS / bandwidth), that results in a back-pressure across all disk writes in the system, and would therefore manifest as one of the above conditions.
 
 ## Write throttling triggers
 
-YugabyteDB defines the following triggers for throttling incoming writes.
+YugabyteDB defines the following triggers for throttling incoming writes:
 
 ### Stop writes trigger
 
 Stop writes trigger is activated in one of the following scenarios:
 
-* **Too many SST files:** The number of SST files exceeds the value determined by the flag `sst_files_hard_limit`, which defaults to 48. Once the hard limit is hit, no more writes are processed, all incoming writes are rejected.
+* **Too many SST files:** The number of SST files exceeds the value determined by the flag `sst_files_hard_limit`, which defaults to 48. After the hard limit is hit, no more writes are processed, all incoming writes are rejected.
+
+* **Memstores flushed too frequently:** This condition occurs if there are a large number of tables (or more accurately, a large number of tablets) all of which get writes. In such cases, the memstores are forced to flush frequently, resulting in too many SST files. In such cases, you can tune the total memstore size allocated. Total memstore size is the minimum of the two flags: `global_memstore_size_mb_max` (default value is 2GB) and `global_memstore_size_percentage` (defaults to 10% of total YB-TServer memory allocated. There are 2 different options for controlling how much memory is allocated to YB-TServer:
```

Comment:
This condition occurs if there are a large number of tables (or more accurately, a large number of tablets)  --> I'm not sure why the same thing needs to be repeated in parentheses

Reply:
Suggestion:

This condition occurs if there is a large number of tables that get writes. In such cases, the memstores are forced to flush frequently, resulting in too many SST files. You can tune the total memstore size allocated. Total memstore size is the minimum of the two flags: `global_memstore_size_mb_max` (default value is 2GB) and `global_memstore_size_percentage` (defaults to 10% of total YB-TServer memory allocated). You can use one of the following options to control the amount of memory allocated to YB-TServer: